### PR TITLE
Backoffice Basic test cases and some optimisation

### PR DESCRIPTION
**Pull request by hardikdiyora92**
@farrukhnaeem-de and @IbeOkoro Kindly have look into this PR. 
- Added Backoffice test case support
- Some code optimisation across different apps automation.

### CODE REVIEW

File: apps/back_office/features/login.feature [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Feature: Check Login Page
+
+  Scenario: Check login page and form
+    Given I am on the login page
+    Then check login form is present
+    When user fill and submit login form
+    Then user can see navigation menu
+    And user can logout from app
```
Comment (farrukhnaeem-de) [inline apps/back_office/features/login.feature]: Comment below for logout refers to this case.

Comment (hardikdiyora92) [reply apps/back_office/features/login.feature]: Hi @farrukhnaeem-de ,
Can you elaborate with logical reason or example of feature file. As I haven't much knowledge in Cucumber.

Current Scenario Outline : Login -> Check/Verify/Validate -> Logout.

Comment (farrukhnaeem-de) [reply apps/back_office/features/login.feature]: Login and logout are two separate entities in terms of their functionality. Both have different purposes which are quite opposite in their nature. One opens up a session and the other closes it. A successful logout has same importance as a login. Which to my understanding deserves a separate scenario to keep it isolated from Login. 

It should be like: 
Login -> Check/Verify/Validate 
Logout -> Check/Validate that the logtout was successfull

Comment (hardikdiyora92) [reply apps/back_office/features/login.feature]: Hi @farrukhnaeem-de ,
In terms of feature file, What do you say about following?

  Background:
    Given I am on the login page
    Then check login form is present
    When user fill and submit login form

  Scenario: Check navigation menu is present
    Then user can see navigation menu

  Scenario: Check user can able to logout successfully
    Then user can logout from app

Comment (farrukhnaeem-de) [reply apps/back_office/features/login.feature]: Background:
Given I am on the login page
Then check login form is present
When user fill and submit login form
Then user can see navigation menu

Making assertions against action in (When) should be done in the respective (Then) not in a separate scenario

Scenario: Check user can able to logout successfully
When user clicks on the Logout button
Then user should be navigate to (whatever page that comes after logout happens)

Comment (hardikdiyora92) [reply apps/back_office/features/login.feature]: Got the point. Incorporated the comment.

**Pull request merged**